truffle/contracts/VaultNotary/VaultNotary.sol

```diff
@@ -0,0 +1,233 @@
+      */
+    function grantUpdateHashPermission(bytes16 vaultId, address anotherAddress)
+        external
+        vaultOwnerOnly(vaultId)
+    {
+        require(!isNotRegistered(vaultId), "vaultId should be registered before grantUpdateHashPermission");
+        notaryMapping[vaultId].aclHashMapping[anotherAddress] = true;
+        emit UpdateHashPermissionGranted(msg.sender, vaultId, anotherAddress);
+    }
+
+    /** @notice Function to revoke update permission to both the Hash field in one vault
+      * @param vaultId The ID of Vault to revoke permission
+      * @param anotherAddress address The address to revoke permission
+      */
+    function revokeUpdateHashPermission(bytes16 vaultId, address anotherAddress)
+        external
+        vaultOwnerOnly(vaultId)
+    {
+        require(!isNotRegistered(vaultId), "vaultId should be registered before revokeUpdateHashPermission");
+        notaryMapping[vaultId].aclHashMapping[anotherAddress] = false;
+        emit UpdateHashPermissionRevoked(msg.sender, vaultId, anotherAddress);
+    }
+
+    /** @notice Function to grant update permission to both Uri field in one vault
+      * @param vaultId bytes16 The ID of Vault to grant permission
+      * @param anotherAddress address The address to grant permission
+      */
+    function grantUpdateUriPermission(bytes16 vaultId, address anotherAddress)
+        external
+        vaultOwnerOnly(vaultId)
+    {
+        require(!isNotRegistered(vaultId), "vaultId should be registered before grantUpdateUriPermission");
+        notaryMapping[vaultId].aclUriMapping[anotherAddress] = true;
+        emit UpdateUriPermissionGranted(msg.sender, vaultId, anotherAddress);
+    }
+
+    /** @notice Function to revoke update permission to both the Uri field in one vault
+      * @param vaultId The ID of Vault to revoke permission
+      * @param anotherAddress address The address to revoke permission
+      */
+    function revokeUpdateUriPermission(bytes16 vaultId, address anotherAddress)
+        external
+        vaultOwnerOnly(vaultId)
+    {
+        require(!isNotRegistered(vaultId), "vaultId should be registered before revokeUpdateUriPermission");
+        notaryMapping[vaultId].aclUriMapping[anotherAddress] = false;
+        emit UpdateUriPermissionRevoked(msg.sender, vaultId, anotherAddress);
+    }
+
+    /** @notice This is used in unit tests to verify the values are correct after using the setters
+      * Can also be used for outside users to read the vaultUri and vaultHash
+      * @param vaultId bytes16 The ID of the Vault to query
+      * @return vaultUri string The uri of the vault
+      * @return vaultHash string The hash of the vault
+      */
+    function getVaultNotaryDetails(bytes16 vaultId)
+        external
+        view
+        returns(string memory vaultUri, string memory vaultHash)
+    {
+        return (notaryMapping[vaultId].vaultUri, notaryMapping[vaultId].vaultHash);
```

Comment:
I can't recall why, but in the existing LoadContract, we don't return a tuple in our getter functions, we just assign each return value (https://github.com/ShipChain/smart-contracts/blob/master/truffle/contracts/Load/LoadContract.sol#L372-L384). I'm not sure if this makes any difference in what gets returned to web3.js when calling the getter. @mlclay do you remember?

Reply:
No, I don't remember why we ended up doing this...

Reply:
https://solidity.readthedocs.io/en/v0.5.11/contracts.html#return-variables

Either are valid. I think I prefer the assignment to the returns variables, instead of returning a tuple. It's more explicit and clear as to which values are being returned for which return variable.